templates/terraform/resource.erb

```diff
@@ -234,8 +234,39 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
     }
 <%    end -%>
 <%    if object.async.is_a? Api::OpAsync -%>
+    <% if object.async.result.resource_inside_response and not object.identity.empty? -%>
```

Comment:
Just to check- is the index id not known until the operation is finished, or is it also contained in the first operation that we then poll? I know firestore_index currently solves this with custom code in the post create, because the name is contained in the first operation response that we already have. Looking at the generated output for this we now set the id on that resource 3 times in create, which isn't ideal. Is there a clean solution that would work for that resource and this?

Reply:
That's right - it's not known until the operation is finished.  :(  I saw that it could be solved with custom code, but I wanted to make sure that I solved it Once And For All - is there an issue with setting the ID three times?  I think we set it once to make sure that we'll do a refresh if we fail in the middle of polling, plus once to get the true final id once all information is known ... and then in firestore, once more due to the custom code solution.  I agree it's not clean, and I'm struggling to test it so I don't even know if it works yet (since you can't use the same project for both firestore and datastore).

Reply:
Got it. I'm pretty ok with it, then. It shouldn't cause issues for users and especially with these templates going away at some point, I don't think it's worth spending a ton of time finding the most-clean-code way to solve this problem for two resources with similar but not identical needs.